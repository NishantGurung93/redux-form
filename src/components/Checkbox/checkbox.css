/* Base for label styling */
[type="checkbox"]:not(:checked),
[type="checkbox"]:checked {
  position: absolute;
  left: -9999px;
}
[type="checkbox"]:not(:checked) + label,
[type="checkbox"]:checked + label {
  position: relative;
  padding-left: 1.95rem;
  cursor: pointer;
  line-height: 1.25rem;
}

/* checkbox aspect */
[type="checkbox"]:not(:checked) + label:before,
[type="checkbox"]:checked + label:before {
  content: '';
  position: absolute;
  left: 0; top: 0;
  width: 1.25em; height: 1.25em;
  border: 2px solid #ccc;
  background: #fff;
  border-radius: 4px;
  box-shadow: inset 0 1px 3px rgba(0,0,0,.1);
}
/* checked mark aspect */
[type="checkbox"]:not(:checked) + label:after,
[type="checkbox"]:checked + label:after {
  content: '\2713\0020';
  position: absolute;
  top: .15rem; left: .22rem;
  font-size: 1.25rem;
  line-height: 0.8;
  transition: all .2s;
}
/* checked mark aspect changes */
[type="checkbox"]:not(:checked) + label:after {
  opacity: 0;
  transform: scale(0);
}
[type="checkbox"]:checked + label:after {
  opacity: 1;
  transform: scale(1);
}

.checkbox-wrapper {
  margin-bottom: 15px;
}
