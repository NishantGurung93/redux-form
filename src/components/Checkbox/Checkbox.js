import { Field } from "formik";
import { string } from 'prop-types';
import './checkbox.css';

const Checkbox = ({ label, name }) => {
  return (
    <div className={'checkbox-wrapper'}>
      <Field
        id={name}
        data-testid={`${name}-checkbox`}
        name={name}
        type='checkbox'
      />
      <label htmlFor={name}>{label}</label>
    </div>
  )
};

Checkbox.propTypes = {
  name: string,
  label: string,
};

export default Checkbox;
