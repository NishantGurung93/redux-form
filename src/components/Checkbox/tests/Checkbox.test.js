import { render } from "@testing-library/react";
import { Formik } from "formik";
import Checkbox from "../Checkbox";

const defaultProps = {
  label: 'Test label',
  name: 'test',
};

const renderComponent = () => {
  return render(
    <Formik>
      <Checkbox {...defaultProps} />
    </Formik>,
  );
};

describe('Input component', () => {
  it('should render the component', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
