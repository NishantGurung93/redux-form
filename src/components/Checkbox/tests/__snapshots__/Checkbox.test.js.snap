// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render the component 1`] = `
<div>
  <div
    class="checkbox-wrapper"
  >
    <input
      data-testid="test-checkbox"
      id="test"
      name="test"
      type="checkbox"
      value=""
    />
    <label
      for="test"
    >
      Test label
    </label>
  </div>
</div>
`;
